.yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/lib/try-path.js b/lib/try-path.js
+index de11ccbf9090658eb0ee61bdba1984ee21bcb3b7..2e75f72483aa7b05bf0a2cde06b3c009eb2103e8 100644
+--- a/lib/try-path.js
++++ b/lib/try-path.js
+@@ -85,6 +85,6 @@ function matchStar(pattern, search) {
+     if (search.substr(search.length - part2.length) !== part2) {
+         return undefined;
+     }
+-    return search.substr(star, search.length - part2.length);
++    return search.substr(star, search.length - part2.length - part1.length);
```

Comment:
I'm not sure if this is a bug or not, but the previous behaviour was unexpected (and I'm not sure how this worked before). Basically when checking a `tsconfig.json` path with a wildcard, this function returns the replacement for the path.

- Pattern: `@metamask/*/node`
  Input: `@metamask/snaps-utils/node`
  Result: `snaps-utils`
- Pattern: `@metamask/*`
  Input: `@metamask/snaps-controllers`
  Result: `snaps-controllers`

And so on. Before this change however, it would return:

- Pattern: `@metamask/*/node`
  Input: `@metamask/snaps-utils/node`
  Result: `snaps-utils/node`

Meaning that it would resolve to `packages/snaps-utils/node/src/...` and some other invalid paths. This causes it to fall back to Node.js' default resolver, which would look at the `package.json` of `snaps-utils` and try to load `packages/snaps-utils/dist/node.js`, but we can't do that since the file may not be built yet.